src/plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -217,7 +217,10 @@ define(function (require) {
             if (rows == null && oldRows == null) return status.LOADING;
 
             var rowsEmpty = _.isEmpty(rows);
-            if (rowsEmpty && fetchStatus) return status.LOADING;
+            // an undefined fetchStatus means the requests are still being
+            // prepared to be sent
+            if (_.isUndefined(fetchStatus)) return status.LOADING;
+            else if (rowsEmpty && fetchStatus) return status.LOADING;
```

Comment:
Hmm, is it expected that `fetchStatus` be something other than `undefined`, like loading, or 0? 

Seems to me perhaps this should be a variable similar to `rowsEmpty` for clarity.